### PR TITLE
feat(react-swatch-picker): changed API of renderingUtils

### PR DESCRIPTION
**Pull request by ValentinaKozlova**
- fixed comments from this PR #30600
- added mixed swatches to stories
- renamed `selectedColor` prop
- added `layout` prop instead of `grid`

### CODE REVIEW

**Comment by fabricteam** [comment]
## 📊 Bundle size report

<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/23e582275522aab677b8ab249f9498d415c0622a'>23e582275522aab677b8ab249f9498d415c0622a</a></sub>
<div id="bundleSizeComment9423"/>

File: packages/react-components/react-swatch-picker-preview/etc/react-swatch-picker-preview.api.md [hunk 1]
```diff
@@ -96,19 +93,30 @@ export const swatchPickerCSSVars: {
...
+};
+
 // @public (undocumented)
 export type SwatchPickerOnSelectEventHandler = EventHandler<SwatchPickerOnSelectionChangeData>;
 
 // @public (undocumented)
 export type SwatchPickerOnSelectionChangeData = EventData<'click', React_2.MouseEvent<HTMLButtonElement>> & {
     selectedValue: string;
-    selectedColor: string;
+    selectedSwatch: string;
```
Comment (ValentinaKozlova) [inline packages/react-components/react-swatch-picker-preview/etc/react-swatch-picker-preview.api.md:113]: I'm thinking whether we need to add separately `selectedColor` and `selectedImage` for cases when colors and used together with images

File: packages/react-components/react-swatch-picker-preview/stories/SwatchPicker/SwatchPickerShape.stories.tsx [hunk 2]
```diff
@@ -26,19 +26,24 @@ export const SwatchPickerShape = () => {
...
+          columnCount: 3,
         })}
       </SwatchPicker>
+      <h3>Rounded</h3>
       <SwatchPicker aria-label="SwatchPicker rounded shape" shape="rounded">
-        {colors.map((color, index) => {
-          return <ColorSwatch key={`${color.value}-${index}`} {...color} />;
+        {colors.map(color => {
+          return <ColorSwatch key={color.value} {...color} />;
         })}
```
Comment (layershifter) [inline packages/react-components/react-swatch-picker-preview/stories/SwatchPicker/SwatchPickerShape.stories.tsx:47]:
```suggestion
        {colors.map(color => <ColorSwatch key={color.value} {...color} />)}
```

nit

File: packages/react-components/react-swatch-picker-preview/stories/SwatchPicker/SwatchPickerLayout.stories.tsx [hunk 3]
```diff
@@ -17,61 +19,83 @@ const useStyles = makeStyles({
...
+        aria-label="SwatchPicker grid layout"
         selectedValue={selectedValue}
         onSelectionChange={handleSelect}
       >
-        {renderSwatchPickerGrid(colors, 3, (row, index) => renderSwatchPickerRow(row, index, ColorSwatchWithTooltip))}
+        {renderSwatchPickerGrid({
+          items,
+          columnCount: 3,
+          renderRow: ({ children, rowId }) => <SwatchPickerRow key={rowId}>{children}</SwatchPickerRow>,
+          renderSwatch: (item: SwatchProps) =>
```
Comment (layershifter) [inline packages/react-components/react-swatch-picker-preview/stories/SwatchPicker/SwatchPickerLayout.stories.tsx:85]: isn't it the same as a default function?

File: packages/react-components/react-swatch-picker-preview/src/utils/renderUtils.tsx [hunk 4]
```diff
@@ -1,31 +1,34 @@
...
   const rows = Array.from({ length: rowCount }, (_, i) => {
     const start = i * columnCount;
     const end = start + columnCount;
-    return colors.slice(start, end);
+    return items.slice(start, end);
   });
 
-  return rows.map(renderRow);
+  return rows.map((row, index) => _renderRow({ children: row.map(_renderSwatch), rowId: index }));
 };
```
Comment (layershifter) [inline packages/react-components/react-swatch-picker-preview/src/utils/renderUtils.tsx:34]:
```ts
function isImageSwatchProps(item: SwatchProps): item is ImageSwatchProps {
  return (item as ImageSwatchProps).src !== undefined;
}

const defaultRenderRow: NonNullable<SwatchPickerGridProps['renderRow']> = ({ children, rowId }) => (
  <SwatchPickerRow key={rowId}>{children}</SwatchPickerRow>
);

const defaultRenderSwatch = (item: SwatchProps) =>
  isImageSwatchProps(item) ? <ImageSwatch {...item} key={item.value} /> : <ColorSwatch {...item} key={item.value} />;

export const renderSwatchPickerGrid = (props: SwatchPickerGridProps) => {
  const { items, columnCount, renderRow = defaultRenderRow, renderSwatch = defaultRenderSwatch } = props;

  const rowCount = Math.ceil(items.length / columnCount);
  const rows = Array.from({ length: rowCount }, (_, i) => {
    const start = i * columnCount;
    const end = start + columnCount;
    return items.slice(start, end);
  });

  return rows.map((row, index) => renderRow({ children: row.map(renderSwatch), rowId: index }));
};
```

What about these changes? 🐱 I think that we will need to find a better way to distinguish between `Image` & `Color`, but let's leave it as it is for now

**Review by layershifter — APPROVED** [review]

File: packages/react-components/react-swatch-picker-preview/stories/SwatchPicker/SwatchPickerLayout.stories.tsx (hunk 3 above)
Comment (ValentinaKozlova) [reply packages/react-components/react-swatch-picker-preview/stories/SwatchPicker/SwatchPickerLayout.stories.tsx:85]: Yeah, I'll remove it

**Review by marcosmoura — APPROVED** [review]

**Pull request merged**